### PR TITLE
Remove kubevirt and set up redirects to community.kubevirt

### LINKED ISSUE #354: Remove collection runtime dependencies
##### SUMMARY
While including other collections for *tests* is fine, including them for runtime is dangerous. Changes in other collections can randomly break things in this collection.

I want to track dependencies to other collections in this issue, with the aim of being able to eventually remove them.

- [x] google.cloud:
  - Runtime:
    - [ ] plugins/lookup/gcp_storage_file.py (uses function `navigate_hash` and class `GcpSession`)
- [x] ansible.netcommon:
  - Runtime:
    - [ ] plugins/modules/net_tools/nios/nios_network.py (uses functions `validate_ip_address`, `validate_ip_v6_address`)
- [x] ansible.posix:
  - Runtime:
    - [x] plugins/modules/system/aix_filesystem.py (uses function `ismount`)
  - Testing only:
    - [ ] tests/unit/plugins/modules/system/interfaces_file/interfaces_file/test_interfaces_file.py
- [x] cisco.intersight:
  - Runtime:
    - [x] plugins/modules/remote_management/intersight/intersight_info.py (uses class `IntersightModule` and constant `intersight_argument_spec` next to docs fragment `cisco.intersight.intersight`)
- [ ] community.kubernetes:
  - Runtime:
    - [ ] plugins/connection/oc.py
    - [ ] plugins/inventory/kubevirt.py
    - [ ] plugins/module_utils/kubevirt.py
    - [ ] plugins/modules/cloud/kubevirt/kubevirt_vm.py
    - [ ] plugins/modules/cloud/kubevirt/kubevirt_rs.py
    - [ ] plugins/modules/cloud/kubevirt/kubevirt_pvc.py
    - [ ] plugins/modules/cloud/kubevirt/kubevirt_template.py
    - [ ] plugins/modules/cloud/kubevirt/kubevirt_cdi_upload.py
    - [ ] plugins/modules/cloud/kubevirt/kubevirt_preset.py
  - Testing only:
    - [ ] tests/unit/plugins/modules/cloud/kubevirt/test_kubevirt_rs.py
    - [ ] tests/unit/plugins/modules/cloud/kubevirt/kubevirt_fixtures.py

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
collection

##### ANSIBLE VERSION
devel

### PR DESCRIPTION
**Pull request by dmsimard**
kubevirt content has been moved to https://github.com/ansible-collections/community.kubevirt

Fixes #354.

### CODE REVIEW

**Comment by felixfontein** [comment]
Todo:
- [ ] add changelog fragment
- [ ] add redirects for doc fragments and module_utils
- [ ] remove community.kubernetes dependency from galaxy.yml
- [ ] remove community.kubernetes dependency from tests/requirements.yml
- [ ] remove community.kubernetes dependency from tests/utils/shippable/shippable.sh

File: changelogs/fragments/1317-kubevirt-migration-removal.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+removed_features:
+  - |
+    All Kubevirt modules and plugins have now been migrated from community.general to the `community.kubevirt <https://galaxy.ansible.com/community/kubevirt>`_ Ansible collection.
+    If you use ansible-base 2.10 or newer, redirections have been provided.
+    If you use Ansible 2.9 and installed this collection, you need to adjust the FQCNs (``community.general.kubevirt_vm`` → ``community.kubevirt.kubevirt_vm``) and make sure to install the community.kubevirt collection.
```
Comment (felixfontein) [inline changelogs/fragments/1317-kubevirt-migration-removal.yml:5]:
```suggestion

    If you use Ansible 2.9 and installed this collection, you need to adjust the FQCNs (``community.general.kubevirt_vm`` → ``community.kubevirt.kubevirt_vm``) and make sure to install the community.kubevirt collection.
```

File: changelogs/fragments/1317-kubevirt-migration-removal.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+breaking_changes:
+  - |
+    If you use Ansible 2.9 and the Kubevirt plugins or modules from this collection, community.general 2.0.0 results in errors when trying to use the Kubevirt content by FQCN, like ``community.general.kubevirt_vm``.
+    Since Ansible 2.9 is not able to use redirections, you will have to adjust your playbooks and roles manually to use the new FQCNs (``community.kubevirt.kubevirt_vm`` for the previous example) and to make sure that you have ``community.kubevirt`` installed.
+    If you use ansible-base 2.10 or newer and did not install Ansible 3.0.0, but installed (and/or upgraded) community.general manually, you need to make sure to also install the ``community.kubevirt`` collection if you are using any of the Kubevirt plugins or modules.
```
Comment (felixfontein) [inline changelogs/fragments/1317-kubevirt-migration-removal.yml:10]:
```suggestion

    If you use ansible-base 2.10 or newer and did not install Ansible 3.0.0, but installed (and/or upgraded) community.general manually, you need to make sure to also install the ``community.kubevirt`` collection if you are using any of the Kubevirt plugins or modules.
```

File: galaxy.yml [hunk 3]
```diff
@@ -7,9 +7,8 @@ authors:
 description: null
 license_file: COPYING
 tags: [community]
-# NOTE: No more dependencies can be added to this list
-dependencies:
-  community.kubernetes: '>=1.0.0'
+# NOTE: No dependencies are expected to be added here
+# dependencies:
```
Comment (felixfontein) [inline galaxy.yml:11]: :tada:

File: meta/runtime.yml [hunk 4]
```diff
@@ -588,3 +593,5 @@ plugin_routing:
       redirect: community.docker.docker_machine
     docker_swarm:
       redirect: community.docker.docker_swarm
+    kubevirt:
+      redirect: community.kubevirt.kubevirt
```
Comment (felixfontein) [inline meta/runtime.yml:597]: There are redirects missing for plugins/module_utils/kubevirt.py, plugins/doc_fragments/kubevirt_common_options.py and plugins/doc_fragments/kubevirt_vm_options.py

Comment (dmsimard) [reply meta/runtime.yml:597]: Good catch.

Comment (dmsimard) [reply meta/runtime.yml:603]: @felixfontein fixed by https://github.com/ansible-collections/community.general/pull/1317/commits/95c858fe45ec13f2b08c4210ff856942935f80ae

**Review by felixfontein — APPROVED** [review]

File: galaxy.yml (hunk 3 above)
Comment (gundalow) [reply galaxy.yml:11]: SUCCESS

**Pull request merged**